garden/migration.md

```diff
@@ -0,0 +1,465 @@
+
+Additionally, take note of the following behaviors:
+
+- Config files are not overwritten, so you might need to delete or manually migrate Ignition-counterpart references in your config files to get them to point to the appropriate Gazebo-counterpart locations
+- Because the install spaces have changed, if you have hard-coded locations involving `ignition`, you might need to migrate them to `gz`
+
+## Migration Steps
+
+It’s important to execute the steps within each of the individual sections in-order! The steps will typically go from specific to general.
+
+### Migrate Files and File References
+
+1. In your package root, look for files and directories matching the (case-insensitive) pattern of `ign(ition)?[_|-]gazebo`, and migrate `ign` / `ignition` appropriately to `gz`, and `gazebo` to `sim`
+2. In your package root, look for files and directories with (case-insensitive) `ign` / `ignition`, and move them to `gz`, matching case
+3. Update all internal references to the migrated files and directories in (1) and (2) in your package
+
+### Migrate CMake
+
+In `CMakeLists.txt` files (and their references in your source files!):
+
+**Variables and macro/function calls**
+
+```cpp
+Find: IGN(ITION)?_GAZEBO
+Replace: GZ_SIM
+
+Find: ign(ition)?_gazebo
+Replace: gz_sim
+
+Find: IGN(ITION)?_
+Replace: GZ_
+
+Find: ign(ition)?_
+Replace: gz_
+```
+
+**Includes**
+
+```cpp
+Find: include\(Ign
+Replace: include(Gz
+
+Find: include\(ign
+Replace: include(gz
+
+Find: gz_find_package\(ign-
+Replace: gz_find_package(gz-
+
+Find: gz_find_package\(Ign(ition)?
+Replace: gz_find_package(Gz-
+```
+
+**Project Names**
+
+```cpp
+Find: ignition-gazebo
+Replace: gz-sim
+
+Find: ignition-
+Replace: gz-
+```
+
+**Note:** Be wary that sometimes CMake arguments trickle down to the source files, so ensure you also migrate them appropriately
+
+### Migrate Macros and Environment Variables
+
+- [Helpful list of env var migrations](https://github.com/gazebo-tooling/release-tools/issues/734)
+- [Helpful list of macro migrations](https://github.com/gazebo-tooling/release-tools/issues/737) (see the toggled drop-down block, some are skipped!)
+
+Migrate source macros and environment variables
+
+```cpp
+Find: IGN(ITION)?_GAZEBO
+Replace: GZ_SIM
+
+Find: ign(ition)?_gazebo
+Replace: gz_sim
+
+Find: IGN(ITION)?_
+Replace: GZ_
+
+Find: ign(ition)?_
+Replace: gz_
+```
+
+For environment variables, you can use the same approach as with macros, but pay attention to the values stored in the environment variables! (e.g. paths.)
+
+Additionally, the logging macros have also been migrated! Migrate any uses!
+
+- `ignerr` -> `gzerr`
+- `ignwarn` -> `gzwarn`
+- `ignmsg` -> `gzmsg`
+- `igndbg` -> `gzdbg`
+- `ignlog` -> `gzlog`
+- `ignLogInit` -> `gzLogInit`
+- `ignLogClose` -> `gzLogClose`
+- `ignLogDirectory` -> `gzLogDirectory`
+
+### Migrate SDF
+
+In `.sdf` files:
+
+```cpp
+Find: <ignition
+Replace: <gz
+
+Find: </ignition
+Replace: </gz
+
+Find: ignition:type
+Replace: gz:type
```

Comment:
This can be more generic `ignition:` -> `gz:`, there are things other instances, like `ignition:expressed_in`

Reply:
https://github.com/gazebosim/docs/pull/275/commits/bbe100c59fd81652c1325a27519af1fe126c4fca